src/main/java/com/google/sps/data/TimeRange.java

```diff
@@ -79,13 +79,14 @@ public boolean overlaps(TimeRange other) {
     //
     // Case 3: |---------|
     //            |---|
+    if (this.end == other.start || other.end == this.start) {
+      return false;
+    }
+
     if (this.start.isBefore(other.start)) {
       return this.contains(other.start);
     }
 
-    if (this.start.equals(other.start)) {
-      return this.end.isAfter(other.end);
-    }
     return other.contains(this.start);
```

Comment:
This function looks pretty good, but can be shortened if you think about what cases would mean no overlap.
There's only no overlap when other.end <= this.start OR other.start >= this.end. So in code:
``` ! (!other.end.isAfter(this.start) || !other.start.isBefore(this.end)) ``` = 
``` other.end.isAfter(this.start) && other.start.isBefore(this.end) ```

Reply:
@apluscs Yes, additional test will be in separate pr